### PR TITLE
Updates to Brave 5.18 with zipkin-reporter 3

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
I think the failures are explainable and will go away in Brave 6, but @anuraaga @reta or @shakuzen if you can check my sanity. What this does is remove the zipkin dependency from *zipkin-reporter*, but brave still references zipkin core types. I think Spring autoconfig scans the whole file then would find those, and so the jar is needed for Spring autoconfig until Brave 6.

Ratpack, cassandra and dropwizard fail because they use third-party tracing which is not updated to not use zipkin types yet. Similar with sleuth.

TL;DR; is I think zipkin-reporter 3 is good, as if it wasn't the other 5 tests wouldn't pass, so we can ship that an move forward to Brave (release 5.18 with reporter 3, but brave still has a zipkin dep for deprecateed methods.. then immediately Brave 6 to remove them)

**Comment by reta** [comment]
> I think the failures are explainable and will go away in Brave 6, but @anuraaga @reta or @shakuzen if you can check my sanity.

I think it makes sense @codefromthecrypt , the 3.x is next major and it is fine to require some work for other projects to migrate (like those deps on zipkin you've mentioned).

**Comment by codefromthecrypt** [comment]
dropwizard-zipkin will be removed as the tracing repo was archived: https://github.com/openzipkin/brave-example/pull/99

**Comment by codefromthecrypt** [comment]
removed ratpack as it was abandoned

File: jersey2-cassandra3/pom.xml [hunk 1]
```diff
@@ -21,15 +21,14 @@
...
     <!-- Use a container that doesn't use Netty. -->
-    <jersey.version>2.33</jersey.version>
-    <spring.version>4.3.29.RELEASE</spring.version>
+    <jersey.version>2.41</jersey.version>
+    <spring.version>4.3.30.RELEASE</spring.version>
 
-    <cassandra.version>3.11.9</cassandra.version>
-    <!-- Use the cassandra-driver-core version bundled in cassandra to not conflict with Netty. -->
-    <cassandra-driver-core.version>3.0.1</cassandra-driver-core.version>
+    <cassandra.version>4.1.3</cassandra.version>
```
Comment (codefromthecrypt) [inline jersey2-cassandra3/pom.xml:27]: was at it, so updated cassandra

File: webflux5-sleuth/pom.xml [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
     <spring-boot.version>2.3.7.RELEASE</spring-boot.version>
     <sleuth.version>2.2.6.RELEASE</sleuth.version>
+    <zipkin-reporter.version>${old-zipkin-reporter.version}</zipkin-reporter.version>
```
Comment (codefromthecrypt) [inline webflux5-sleuth/pom.xml:24]: sleuth uses an old version of zipkin-reporter and as it is unmaintained, probably won't update. Brave 6 removes version management from reporter for reasons like this.
